ndmtestpackage.py

```diff
@@ -0,0 +1,386 @@
+
+NDM_YAML = 'node-disk-manager.yaml'
+NDM_TEST_YAML_PATH = '/tmp/'
+NDM_TEST_YAML_NAME = 'NDM_Test_'+NDM_YAML
+DOCKER_IMAGE_NAME = 'openebs/node-disk-manager'
+try:
+    DOCKER_IMAGE_TAG = subprocess.check_output(["git",
+                                                "rev-parse",
+                                                "--abbrev-ref",
+                                                "HEAD"]).strip()
+except subprocess.CalledProcessError as err:
+    print 'Subprocess error occured while getting HEAD name:', err.returncode
+    raise err
+except Exception as err:
+    print 'Unknown error occured while getting HEAD name.'
+    raise err
+
+# Different phases of Pod
+POD_WAIT_PHASES = ['Pending']
+POD_GOOD_PHASES = ['Running']
+POD_BAD_PHASES = ['Error']
+
+# Different states of Pod
+POD_WAIT_STATES = ['ContainerCreating', 'Pending']
+POD_GOOD_STATES = ['Running']
+POD_BAD_STATES = ['CrashLoopBackOff', 'ImagePullBackOff']
+
+class NDMTestException(Exception):
+    pass
+
+#def validateNDMPod(ndmPod):
+
+def validate_ndm_log(log):
+    # Definition of this function should grow as program grows
+    if 'Started controller' in log:
+        return True
+    return False
+
+def get_ndm_pod():
+    config.load_kube_config()
+    v1 = client.CoreV1Api()
+
+    # Try to get node-disk-manager for 5 times as sometime code reaches
+    # when pod is not even in ContainerCreating state
+    i = 0
+    ndmPod = None
+    while ndmPod == None and i < 5:
+        sleep(1)
+
+        # List pods
+        # Assumption: NDM pod runs under 'dafault' namespace.
+        pods = v1.list_namespaced_pod('default')
+
+        # Find NDM Pod
+        for pod in pods.items:
+            #Assumption: Pod name starts with string 'node-disk-manager'.
+            if pod.metadata.name.startswith('node-disk-manager'):
+                ndmPod = pod
+                break
+        i += 1
+    else:
+        if ndmPod == None:
+            print 'Failed getting NDM-Pod.'
+            exit(1)
+
+    return ndmPod
+
+def get_pod_phase(pod):
+    return pod.status.phase
+
+def get_container_state_in_ndm_pod(container_index=0, wait_for_sec=10):
+    waited = 0
+    pod = get_ndm_pod()
+    while pod.status.container_statuses is None and waited < wait_for_sec:
+        sleep(1)
+        waited += 1
+        pod = get_ndm_pod()
+    if not waited < wait_for_sec:
+        raise NDMTestException('Pod had no container till '\
+                               + str(wait_for_sec) + ' seconds.')
+
+    while len(pod.status.container_statuses) <= container_index\
+            and waited < wait_for_sec:
+        sleep(1)
+        waited += 1
+        pod = get_ndm_pod()
+    if not waited < wait_for_sec:
+        raise NDMTestException('Pod did not had ' + str(container_index+1)\
+                               + ' containers till ' + str(wait_for_sec)\
+                               + ' seconds.')
+
+    return pod.status.container_statuses[container_index].state
+
+def make():
```

Comment:
Let us remove the make functionality from this. It will be invoked directly as a target.

Reply:
I am OK with whatever is the best practice or whatever align best with our use-cases.